pulp_rpm/app/models.py

```diff
@@ -734,23 +745,83 @@ class PackageGroup(Content):
     default = models.BooleanField(default=False)
     user_visible = models.BooleanField(default=False)
 
-    display_order = models.IntegerField()
+    display_order = models.IntegerField(null=True)
     name = models.CharField(max_length=255)
-    description = models.TextField()
-    packages = models.TextField()
+    description = models.TextField(default='')
+    packages = JSONField(default=list)
 
     biarch_only = models.BooleanField(default=False)
 
-    desc_by_lang = models.TextField()
-    name_by_lang = models.TextField()
+    desc_by_lang = JSONField(default=dict)
+    name_by_lang = JSONField(default=dict)
 
     digest = models.CharField(unique=True, max_length=64)
 
+    related_packages = models.ManyToManyField(Package)
+
     class Meta:
         default_related_name = "%(app_label)s_%(model_name)s"
 
+    @classmethod
+    def natural_key_fields(cls):
+        """
+        Digest is used as a natural key for PackageGroups.
+        """
+        return ('digest',)
+
+    @classmethod
+    def pkglist_to_lst(cls, value):
+        """
+        Convert libcomps PkgList to list.
+
+        Args:
+            value: a libcomps PkgList
+
+        Returns:
+            A list
+
+        """
+        package_list = []
+        if len(value):
```

Comment:
Does it complain if you don't check the len of value?
Usually  you can use for and it won't go into the loop  if there is nothing to iterate.

Reply:
It does not appear to complain upon removal, so removed it.